app/javascript/src/local_level.tsx

```diff
@@ -84,11 +112,20 @@ class Tile extends React.Component {
   }
 }
 
-class LocalLevel extends React.Component {
-  constructor(props) {
+interface LocalLevelState {
+  messages: string[];
+  errors:   string[];
+  tiles:    ITile[][];
+  player:   IPlayer;
+}
+
+class LocalLevel extends React.PureComponent<{ data: LocalLevelState }> {
```

Comment:
May I suggest to export the class right away instead of exporting a default. That way it's more consistent with places where you want to export more than one component or prop types eg `import {LocalLevel, LocalLevelProps} from './local_level'`